Sources/SnowManLogicSpecs/ViewSpec.swift

```diff
@@ -0,0 +1,106 @@
+                        	io.stubbedUserInput = ["?a", "fail"]
+                        	expect(view.receiveGuess(io.myReadLine)).to.equal("fail")
+                    	}
+
+                    	it("should return an error message if the user hits return") {
+                        	io.stubbedUserInput = ["", "fail"]
+                        	expect(view.receiveGuess(io.myReadLine)).to.equal("fail")
+                    	}
+        	}
+
+        	describe("#assignBlanks") {
+            		var gameWord: String!
+            		var correctGuesses: [String]!
+            
+            		before() {
+                		gameWord = "hello"
+                		correctGuesses = []
+            		}
+            
+            		it("should return only blanks for unguessed word") {
+                		expect(view.assignBlanks(gameWord, correctGuesses: correctGuesses)).to.equal("__  __  __  __  __ ")
+            		}
+
+            		it("should return no blanks if the word has been guessed") {
+                		correctGuesses = ["h", "e", "l", "o"]
+                		expect(view.assignBlanks(gameWord, correctGuesses: correctGuesses)).to.equal("h e l l o")
+            		}
+
+            		it("should return blanks and letters for partially guessed word") {
+                		correctGuesses = ["h", "e"]
+                		expect(view.assignBlanks(gameWord, correctGuesses: correctGuesses)).to.equal("h e __  __  __ ")
+            		}
+
+        	}
+
+	        describe("#displayRemainingGuesses") {
+        		it("should return the number of guesses remaining") {
+                    		var remainingGuesses = 4
+                    		expect(view.displayRemainingGuesses(4)).to.equal("You have 4 remaining guesses")
+                
+                    		remainingGuesses = 0
+                    		expect(view.displayRemainingGuesses(0)).to.equal("You have 0 remaining guesses")
+
+                    		remainingGuesses = -2
+                    		expect(view.displayRemainingGuesses(-2)).to.equal("You have -2 remaining guesses")
+
+                	}
+            	}
+
+            	describe("#displayResults") {
+                	var gameWord: String!
+
+                	before() {
+                    		gameWord = "hello"
+                	}
+                
+			it("should print a congragulatory message if winning conditions met") {
+                    		let winner = true
+                    		expect(view.displayResults(gameWord, winner: winner)).to.equal("Congratulations. You win! You correctly guessed \"hello\"")
+                	}
+
+                	it("should inform player of loss if winning conditions not met") {
+                    		let winner = false
+                    		expect(view.displayResults(gameWord, winner: winner)).to.equal("Game Over. You ran out of guesses. The word was \"hello\"")
+                	}
+            	}
+  	}
+}
+
+private func myUserInput() -> String {
```

Comment:
Removed. I was previously using it to test the test. My `MockIO`, `MockView` and Associated tests have been exploded.